In [1]:
suppressWarnings(suppressPackageStartupMessages({
    library(tidyverse)
    library(data.table)
}))


In [23]:
# input
pvar_f   <- '/oak/stanford/groups/mrivas/ukbb24983/cnv/pgen/cnv.pvar'
csq_rules_f <- '../VEP_consequence_group.tsv'
vep_f    <- '/scratch/groups/mrivas/ukbb24983/cnv/annotation_20201003/cnv.vep101-loftee.20201009.tsv.gz'
# output
vep_csq_f <- str_replace(vep_f, '.tsv.gz$', '.Csq.tsv')


In [3]:
csq_rules_f %>% fread(select=c('#Consequence', 'Csq')) %>%
rename('Consequence'='#Consequence') -> csq_rules_df


In [4]:
vep_f %>% fread(colClasses = c('#CHROM'='character')) %>%
rename('CHROM'='#CHROM') -> vep_df


In [16]:
pvar_f %>% fread(colClasses = c('#CHROM'='character')) %>% rename('CHROM'='#CHROM') %>%
separate(ID, c(NA, 'POS_s', 'POS_e', NA), remove=F) %>%
mutate(pvar_order = 1:n()) -> pvar_df


In [25]:
pvar_df %>%
left_join(vep_df %>% select(-CHROM, -POS), by='ID') %>%
left_join(csq_rules_df, by='Consequence') %>%
arrange(as.numeric(CHROM), as.numeric(POS_s), as.numeric(POS_e), pvar_order) %>%
select(-FILTER) -> df


In [26]:
df %>%
rename('#CHROM' = 'CHROM') %>%
fwrite(vep_csq_f, sep='\t', na = "NA", quote=F)


## check the `Csq` assignments

In [21]:
df %>% count(Csq)

Csq    n     
1 intron  54188
2 others  64440
3 pav      1143
4 pcv       774
5 ptv    152268
6 utr      2117
7 NA        250

We do not have variant annotation for 250 variants.


In [22]:
vep_df %>%
count(Consequence) %>%
left_join(csq_rules_df, by='Consequence') %>%
arrange(Csq, -n)

Consequence                        n      Csq   
1  intron_variant                      54188 intron
2  intergenic_variant                  24390 others
3  TFBS_ablation                       16525 others
4  upstream_gene_variant                7583 others
5  downstream_gene_variant              5395 others
6  regulatory_region_ablation           3974 others
7  non_coding_transcript_exon_variant   3458 others
8  regulatory_region_variant            2880 others
9  TF_binding_site_variant               222 others
10 mature_miRNA_variant                   13 others
11 splice_region_variant                1073 pav   
12 inframe_insertion                      41 pav   
13 inframe_deletion                       18 pav   
14 protein_altering_variant               11 pav   
15 coding_sequence_variant               772 pcv   
16 stop_retained_variant                   2 pcv   
17 transcript_ablation                103617 ptv   
18 splice_acceptor_variant             24930 ptv   
19 stop_gained                         17867 ptv   
20 splice_donor_variant                 4102 ptv   
21 frameshift_variant                   1422 ptv   
22 stop_lost                             326 ptv   
23 start_lost                              4 ptv   
24 3_prime_UTR_variant                  1511 utr   
25 5_prime_UTR_variant                   606 utr